pandas/core/frame.py

```diff
@@ -3000,7 +3004,10 @@ def set_index(self, keys, drop=True, append=False, inplace=False,
                     to_remove.append(col)
             arrays.append(level)
 
-        index = MultiIndex.from_arrays(arrays, names=names)
```

Comment:
same (or at least that's the intent); otherwise maybe should have an option / another helper function to avoid this repetition of code.